src/mcts/node.cc

```diff
@@ -523,6 +524,16 @@ bool NodeTree::ResetToPosition(const std::string& starting_fen,
   // retain old n_ and q_ (etc) data, even though its old children were
   // previously trimmed; we need to reset current_head_ in that case.
   if (!seen_old_head) TrimTreeAtHead();
+  // When reusing the tree, the new head might have some two-fold terminals that
+  // don't actually end the game, so make them regular nodes to extend.
+  else if (current_head_->HasChildren()) {
+    for (const auto& child : current_head_->Edges()) {
+      const auto node = child.node();
+      if (node && node->terminal_type_ == Node::Terminal::TwoFold) {
+        node->MakeNotTerminal();
```

Comment:
This optimization does an eager 1-depth reverting of twofold terminals but more correct would be to do it for all terminals in the tree, but finding them with tree traversal can be expensive. One idea is to lazily find them during `PickNodeToExtend` as it's traversing the tree anyway and only a smaller subset are terminal and even smaller subset are twofold, so even if it's not trivial to check, it should be infrequent. This will probably need some helper like the one in #682 as the node doesn't have the position and needs to be built like `ExtendNode`

Reply:
A general observation - MakeNotTerminal reverts to 0 visits, but it doesn't update the parents to remove those N terminal 0.0 visits when it does so. In order to give appropriate exploration focus on this choice that is no longer a draw, that backup should be  performed.
In this current CL it doesn't matter too much because it only happens for root children - but once you expand it further down the tree it'll start to matter much more.